app/draft.go

```diff
@@ -39,7 +39,7 @@ func (a *App) UpsertDraft(c *request.Context, draft *model.Draft, connectionID s
 		return nil, model.NewAppError("UpsertDraft", "app.draft.feature_disabled", nil, "", http.StatusNotImplemented)
 	}
 
-	dt, dErr := a.Srv().Store().Draft().Get(draft.UserId, draft.ChannelId, draft.RootId)
+	dt, dErr := a.Srv().Store().Draft().Get(draft.UserId, draft.ChannelId, draft.RootId, true)
```

Comment:
Do we really need to update a _deleted_ draft? I think a check can be added which may return a different error in case you're trying to update a deleted draft?

Reply:
@vish9812 drafts have channel-id & user-id as the primary key, so we need to update a soft deleted draft.
But I think we could just delete the row upon delete, and then insert a new one when creating one, (and remove the deleteat column).

Reply:
Oh okay, if there is only ever gonna be one draft per channel then the 2nd case makes sense to me, since we are not gonna keep the deleted drafts anyway, in case of updates. Also, the deleted draft may not be of any use.